[FIX] we need to parse the parameter to an int

hbrunn opened the pull request:
without this, configuring '0' triggers an infinite loop on some browsers

hbrunn committed:
[FIX] we need to parse the parameter to an int

tarteo commented:
@hbrunn Is the if statement triggering an infinite loop or the setTimeout when timeout is '0'? If it's the setTimeout triggering it you'll need to reset the variable. like this `timeout = parseInt(timeout)`

hbrunn commented:
the timeout. In the conditional, `'0'` is coerced to boolean `true`, in `setTimeout`, it's coerced to int `0`, and then the browser freezes

tarteo reviewed (approved):
@hbrunn Thanks, I understand the problem now.

hparfr commented:
If the intent is to trigger a callback periodically, [setInterval()](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setInterval) should be used instead.

yajo committed:
Explicit radix

According to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#Parameters, we should always specify the radix.
